Source/Core/DolphinQt2/Resources.cpp

```diff
@@ -2,46 +2,64 @@
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
+#include <algorithm>
+
+#include <QGuiApplication>
+#include <QScreen>
 #include <QStringList>
 
 #include "Common/CommonPaths.h"
 #include "Common/FileUtil.h"
 #include "DolphinQt2/Resources.h"
+#include "DolphinQt2/Settings.h"
 
 QList<QPixmap> Resources::m_platforms;
 QList<QPixmap> Resources::m_countries;
 QList<QPixmap> Resources::m_ratings;
 QList<QPixmap> Resources::m_misc;
 
+QString Resources::GetPreferredPixmapScale(int max)
```

Comment:
You probably missed it in the shuffle, I asked on an earlier commit: Instead of `max`, why not always try to load the best pixel ratio image, and fallback if it doesn't exist, just like QIcon does internally?